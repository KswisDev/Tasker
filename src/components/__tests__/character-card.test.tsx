import { render } from "@testing-library/react";
import React from "react";
import CharacterCard from "../character-card";

describe("CharacterCard", () => {
  it("renders as expected", async () => {
    expect(
      render(
        <CharacterCard
          character={{
            name: "Rick",
            id: 1,
            image: "url",
            species: "human",
            status: "alive",
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
