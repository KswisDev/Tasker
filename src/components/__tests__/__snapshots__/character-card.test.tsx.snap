// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard renders as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mr-6 mb-4 border-2 border-gray-600 text-center"
      >
        <h2>
          Rick
        </h2>
        <img
          alt="picture of url"
          height="300"
          src="url"
          width="300"
        />
        <div>
          <span
            class="font-bold"
          >
            Status:
          </span>
           
          alive
        </div>
        <div>
          <span
            class="font-bold"
          >
            Species:
          </span>
           
          human
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mr-6 mb-4 border-2 border-gray-600 text-center"
    >
      <h2>
        Rick
      </h2>
      <img
        alt="picture of url"
        height="300"
        src="url"
        width="300"
      />
      <div>
        <span
          class="font-bold"
        >
          Status:
        </span>
         
        alive
      </div>
      <div>
        <span
          class="font-bold"
        >
          Species:
        </span>
         
        human
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterCard renders loading as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mr-6 mb-4 border-2 border-gray-600 text-center"
      >
        <h2>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
            >
              ‌
            </span>
            <br />
          </span>
        </h2>
        <div
          class="w-[300px] h-[300px]"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="height: 100%;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div>
          <span
            class="font-bold"
          >
            Status:
          </span>
           
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div>
          <span
            class="font-bold"
          >
            Species:
          </span>
           
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mr-6 mb-4 border-2 border-gray-600 text-center"
    >
      <h2>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
        </span>
      </h2>
      <div
        class="w-[300px] h-[300px]"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="height: 100%;"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div>
        <span
          class="font-bold"
        >
          Status:
        </span>
         
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="width: 50%;"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div>
        <span
          class="font-bold"
        >
          Species:
        </span>
         
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="width: 50%;"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
